src/python/pants/rules/core/test.py

```diff
@@ -25,25 +27,25 @@ class Test(Goal):
 @console_rule(Test, [Console, BuildFileAddresses])
 def fast_test(console, addresses):
   test_results = yield [Get(TestResult, Address, address.to_address()) for address in addresses]
-  wrote_any_stdout = False
   did_any_fail = False
-  for test_result in test_results:
-    wrote_any_stdout |= bool(test_result.stdout)
-    # Assume \n-terminated
-    console.write_stdout(test_result.stdout)
-    if test_result.stdout and not test_result.stdout[-1] == '\n':
-      console.write_stdout('\n')
+  for address, test_result in zip(addresses, test_results):
+    if test_result.stdout:
+      console.write_stdout(
```

Comment:
These is an existing global flag that should be incorporated here:
```
--[no-]colors (default: True)
    Set whether log messages are displayed in color.
```

It's possible that it could be fed into the creation of the ... `Console` object without increasing the scope here too much?

Reply:
...here: https://github.com/pantsbuild/pants/blob/3742edb4040967899c404b43570caa721bb3bd95/src/python/pants/init/engine_initializer.py#L183

(and then probably in some unit tests as well, which are creating `MockConsole` objects.)